metaflow/plugins/airflow/airflow_utils.py

```diff
@@ -0,0 +1,410 @@
+    _arg_types = {
+        "dag_id": str,
+        "description": str,
+        "schedule_interval": str,
+        "start_date": datetime,
+        "catchup": bool,
+        "tags": list,
+        "dagrun_timeout": timedelta,
+        "default_args": {
+            "owner": str,
+            "depends_on_past": bool,
+            "email": list,
+            "email_on_failure": bool,
+            "email_on_retry": bool,
+            "retries": int,
+            "retry_delay": timedelta,
+            "queue": str,  #  which queue to target when running this job. Not all executors implement queue management, the CeleryExecutor does support targeting specific queues.
+            "pool": str,  # the slot pool this task should run in, slot pools are a way to limit concurrency for certain tasks
+            "priority_weight": int,
+            "wait_for_downstream": bool,
+            "sla": timedelta,
+            "execution_timeout": timedelta,
+            "trigger_rule": str,
+        },
+    }
+
+    # Reference for user_defined_filters : https://stackoverflow.com/a/70175317
+    filters = dict(
+        task_id_creator=lambda v: task_id_creator(v),
+        json_dump=lambda val: json_dump(val),
+        run_id_creator=lambda val: run_id_creator(val),
+    )
+
+    def __init__(self, **kwargs):
+        self._args = kwargs
+
+    @property
+    def arguments(self):
+        return dict(**self._args, user_defined_filters=self.filters)
+
+    def serialize(self):
+        def parse_args(dd):
+            data_dict = {}
+            for k, v in dd.items():
+                # see the comment below for `from_dict`
+                if isinstance(v, dict):
+                    data_dict[k] = parse_args(v)
+                elif isinstance(v, datetime):
+                    data_dict[k] = v.isoformat()
+                elif isinstance(v, timedelta):
+                    data_dict[k] = dict(seconds=v.total_seconds())
+                else:
+                    data_dict[k] = v
+            return data_dict
+
+        return parse_args(self._args)
+
+    @classmethod
+    def deserialize(cls, data_dict):
+        def parse_args(dd, type_check_dict):
+            kwrgs = {}
+            for k, v in dd.items():
+                if k not in type_check_dict:
+                    kwrgs[k] = v
+                    continue
+                if isinstance(v, dict) and isinstance(type_check_dict[k], dict):
+                    kwrgs[k] = parse_args(v, type_check_dict[k])
+                elif type_check_dict[k] == datetime:
+                    kwrgs[k] = datetime.fromisoformat(v)
+                elif type_check_dict[k] == timedelta:
+                    kwrgs[k] = timedelta(**v)
+                else:
+                    kwrgs[k] = v
+            return kwrgs
+
+        return cls(**parse_args(data_dict, cls._arg_types))
+
+
+def _kubernetes_pod_operator_args(operator_args):
```

Comment:
TODO (savin): Review that this is functionally equivalent to `argo-workflows` and `kubernetes`